qcodes/tests/drivers/test_m2j.py

```diff
@@ -0,0 +1,35 @@
+import sys
+import unittest
+from unittest.mock import patch, MagicMock, call
+
+sys.modules['spirack'] = MagicMock(name='spirack')
+from spirack import M2j_module, SPI_rack
+from qcodes.instrument_drivers.QuTech.M2j import M2j
+
+
+class TestZIHDAWG8(unittest.TestCase):
+
+    def test_gain(self):
+        spi_rack = MagicMock()
+        m2j = M2j('test', spi_rack, 42)
+
+        with self.assertRaises(ValueError):
+            m2j.gain.set(32)
+        with self.assertRaises(ValueError):
+            m2j.gain.set(56)
+
+        for gain in range(33, 56):
+            m2j.gain.set(gain)
```

Comment:
If you test many values to be set, then also set at least 1 float instead of only integers.